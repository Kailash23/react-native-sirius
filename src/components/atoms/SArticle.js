import React from 'react';
import { Title, withTheme } from 'react-native-paper';
import { customFonts } from '@styles/customFonts';
import { SIZES, getFontSize } from '@styles/sizes';

function SArticle(props) {
  const {
    size,
    title,
    paddingTop,
    paddingBottom,
    style,
    lineHeight,
    avoidTextScaling,
    fontFamily,
    color,
    numberOfLines
  } = props;

  function scaledText(size) {
    if (avoidTextScaling) {
      return size;
    } else {
      return getFontSize(size);
    }
  }
  return (
    title && (
      <Title
        numberOfLines={numberOfLines && numberOfLines}
        style={[
          {
            fontFamily: fontFamily || customFonts.torstarTextO2Roman,
            fontSize: size ? scaledText(size) : scaledText(SIZES.BASE),
            lineHeight: lineHeight
              ? scaledText(lineHeight)
              : scaledText(SIZES.BASE + 4),
            paddingTop: paddingTop ? paddingTop : 0,
            paddingBottom: paddingBottom ? paddingBottom : 0,
            color: color || props.theme.colors.text
          },
          { ...style }
        ]}>
        {title}
      </Title>
    )
  );
}

export default withTheme(SArticle);
