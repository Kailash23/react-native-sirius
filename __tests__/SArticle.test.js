import SArticle from '@atoms/SArticle';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <SArticle/>', () => {
  test('SArticle should render properly with given styles and title', () => {
    const wrapper = shallow(
      <SArticle
        title={'Test Article'}
        size={16}
        paddingBottom={20}
        lineHeight={18}
        color={'white'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('SArticle should NOT render if the title is null', () => {
    const wrapper = shallow(
      <SArticle
        title={null}
        size={16}
        paddingBottom={20}
        lineHeight={18}
        color={'white'}
      />
    );
    expect(wrapper).toEqual({});
  });
});
