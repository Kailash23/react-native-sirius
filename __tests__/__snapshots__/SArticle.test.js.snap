// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SArticle/> SArticle should render properly with given styles and title 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  color="white"
  lineHeight={18}
  paddingBottom={20}
  size={16}
  title="Test Article"
/>
`;
